src/Smearing.jl

```diff
@@ -94,46 +97,44 @@ end
 
 struct Gaussian <: SmearingFunction end
 occupation(S::Gaussian, x) = (1 - erf(x)) / 2
-entropy(S::Gaussian, x) = 1 / (2sqrt(typeof(x)(pi))) * exp(-x^2)
-
-H1(x) = 2x
-H2(x) = 4x^2 - 2
-H3(x) = 8x^3 - 12x
-H4(x) = 16x^4 - 48x^2 + 12
-A(n, T=Float64) = (-1)^n / (factorial(n) * 4^n * sqrt(T(π)))
-
-struct MethfesselPaxton1 <: SmearingFunction end
-function occupation(S::MethfesselPaxton1, x)
-    x == Inf && return zero(x)
-    x == -Inf && return one(x)
-    occupation(Gaussian(), x) + A(1, typeof(x))*H1(x)*exp(-x^2)
+entropy(S::Gaussian, x::T) where T = 1 / (2 * sqrt(T(pi))) * exp(-x^2)
+
+struct MarzariVanderbilt <: SmearingFunction end
+occupation(S::MarzariVanderbilt, x::T) where T = 2 - erfc(x - 1/sqrt(T(2))) + sqrt(2 / T(π)) * exp(-x^2 + sqrt(T(2)) * x - 1/T(2))
+entropy(S::MarzariVanderbilt, x::T) where T = 1 / sqrt(2 * T(π)) * (x - 1/sqrt(T(2))) * exp(-x^2)
+
+A(T::Type, n) = (-1)^n / (factorial(n) * 4^n * sqrt(T(π)))
+function H(x, n)
+    if iszero(n)
+        return one(x)
+    elseif isone(n)
+        return 2 * x
+    else
+        return 2 * x * H(x, n-1) - 2 * (n-1) * H(x, n-2)
+    end
 end
-entropy(S::MethfesselPaxton1, x) = 1/2 * A(1, typeof(x)) * H2(x) * exp(-x^2)
 
-struct MethfesselPaxton2 <: SmearingFunction end
-function occupation(S::MethfesselPaxton2, x)
-    x == Inf && return zero(x)
-    x == -Inf && return one(x)
-    occupation(MethfesselPaxton1(), x) + A(2, typeof(x))*H3(x)*exp(-x^2)
+struct MethfesselPaxton <: SmearingFunction
+    order::Int
 end
-entropy(S::MethfesselPaxton2, x) = 1/2 * A(2, typeof(x)) * H4(x) * exp(-x^2)
-
-function MethfesselPaxton(order::Integer)
-    if order == 0
-        Gaussian()
-    elseif order == 1
-        MethfesselPaxton1()
-    elseif order == 2
-        MethfesselPaxton2()
-    else
-        error("Not implemented")
+function occupation(S::MethfesselPaxton, x::T) where T
+    S₀ = 1/T(2) * erfc(-x)
+    ΣSₙ = zero(x)
+    for i in 1:S.order
+        ΣSₙ -= A(T, i) * H(x, 2i - 1)
     end
+    return S₀ + ΣSₙ * exp(-x^2)
+end
+function entropy(S::MethfesselPaxton, x::T) where T
+    ΣSₙ = zero(x)
+    for i in 0:S.order
+        ΣSₙ -= A(T, i) * H(x, 2i + 1)
+    end
+    return ΣSₙ * exp(-x^2)
 end
-
-# TODO: Marzari-Vanderbilt "cold smearing"
 
 # List of available smearing functions
-smearing_methods = (None, FermiDirac, Gaussian, MethfesselPaxton1, MethfesselPaxton2)
+smearing_methods = (None, FermiDirac, Gaussian, MarzariVanderbilt, MethfesselPaxton)
```

Comment:
that won't work (see test/occupation.jl which calls `method()`). The simplest is probably to change this to
```
smearing_methods = (None(), FermiDirac(), Gaussian(), MarzariVanderbilt(), MethfesselPaxton(1), MethfesselPaxton(2))
```
and make corresponding changes to test/occupation.jl. And maybe just move that list to test/occupation.jl anyway...

Reply:
Yes, this was the first issue with the tests, I've (hopefully) fixed it in the latest commit. Thanks!